indico/modules/events/registration/logging.py

```diff
@@ -0,0 +1,49 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2020 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+from flask import session
+
+from indico.core import signals
+from indico.modules.events import EventLogKind, EventLogRealm
+from indico.modules.events.registration.models.registrations import RegistrationState
+from indico.util.i18n import orig_string
+
+
+def connect_log_signals():
+    signals.event.registration_checkin_updated.connect(log_registration_check_in)
+    signals.event.registration_state_updated.connect(log_registration_updated)
+
+
+def log_registration_check_in(registration, **kwargs):
+    """Log a registration check-in action to the event log."""
+    if registration.checked_in:
+        log_text = '"{}" has been checked in'.format(registration.full_name)
+    else:
+        log_text = '"{}" check-in has been reset'
+    registration.event.log(EventLogRealm.participants, EventLogKind.change, 'Registration',
+                           log_text.format(registration.full_name), session.user)
+
+
+def log_registration_updated(registration, previous_state, **kwargs):
+    """Log the registration status change to the event log."""
+    from indico.modules.events.registration import logger
+    if not previous_state:
+        return
+    if registration.state == RegistrationState.complete:
+        log_text = 'Registration for "{}" has been approved'
+    elif registration.state == RegistrationState.rejected:
+        log_text = 'Registration for "{}" has been rejected'
+    elif registration.state == RegistrationState.pending:
+        previous_state_title = orig_string(previous_state.title).lower()
+        log_text = 'Registration for "{}" has been changed from %s to pending' % previous_state_title
+    else:
+        logger.error('Unknown registration state "%s" for registration "%s"', (registration.state, registration))
```

Comment:
"not enough format params" because loggers take varargs instead of a single args for the format params